cartographer/metrics/gauge.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_METRICS_GAUGE_H_
+#define CARTOGRAPHER_METRICS_GAUGE_H_
+
+#include <map>
+#include <vector>
+
+namespace cartographer {
+namespace metrics {
+
+class Gauge {
```

Comment:
There was some review discussion about typing Gauge. Probably best with templates: Gauge<bool>, Gauge<int>, Gauge<double>. Should we implement this right away? On the other hand, the Prometheus implementation can't differentiate between them anyway and double should have enough precision to store bools and ints as well, so it's not terribly important.

Reply:
I would rather wait until at least one implementation handles it.